compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -428,18 +428,16 @@ public void visit(BLangTypeDefinition astTypeDefinition) {
                                                           displayName,
                                                           astTypeDefinition.symbol.originalName);
         if (astTypeDefinition.symbol.tag == SymTag.TYPE_DEF) {
-            if (type.tsymbol.owner == astTypeDefinition.symbol.owner) {
-                typeDefs.put(astTypeDefinition.symbol.type.tsymbol, typeDef);
-                typeDef.referenceType = ((BTypeDefinitionSymbol) astTypeDefinition.symbol).referenceType;
+            BTypeReferenceType referenceType = ((BTypeDefinitionSymbol) astTypeDefinition.symbol).referenceType;
+            typeDef.referenceType = referenceType;
+            BTypeSymbol typeSymbol = astTypeDefinition.symbol.type.tsymbol;
+            if (type.tsymbol.owner == astTypeDefinition.symbol.owner
+                    && !(Symbols.isFlagOn(typeSymbol.flags, Flags.CLASS))) {
```

Comment:
Why do we need to do this at all? Even if we do have type reference types at runtime, for something like

```ballerina
type Foo Bar;
```

shouldn't we just be doing

```java
$type$Foo = $type$Bar;
```

Reply:
We can revisit this separately.

Reply:
The reference type will not be created at runtime. AFAIR, this was to filter out the definition created for the class.